src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -844,8 +844,11 @@ private void doHandleAnonymous(HttpServletRequest request,
     }
 
     private void handleGetContainerAcl(HttpServletResponse response,
-            BlobStore blobStore, String containerName) throws IOException {
-        ContainerAccess access = blobStore.getContainerAccess(containerName);
+            BlobStore blobStore, String containerName)
+            throws IOException, S3Exception {
+        if (!blobStore.containerExists(containerName)) {
+            throw new S3Exception(S3ErrorCode.NO_SUCH_BUCKET);
+        }
```

Comment:
Can you add back the `getContainerAccess` call below the `containerExists`?  This minimizes the diff and avoids possible exceptions while writing the response to the client.